src/Adapter/News/NewsDataProvider.php

```diff
@@ -0,0 +1,173 @@
+     * @param CountryDataProvider $countryDataProvider
+     * @param Tools $tools
+     * @param Configuration $configuration
+     * @param Validate $validate
+     * @param int $contextMode
+     */
+    public function __construct(
+        CircuitBreakerInterface $circuitBreaker,
+        CountryDataProvider $countryDataProvider,
+        Tools $tools,
+        Configuration $configuration,
+        Validate $validate,
+        $contextMode
+    ) {
+        $this->circuitBreaker = $circuitBreaker;
+        $this->configuration = $configuration;
+        $this->contextMode = $contextMode;
+        $this->countryDataProvider = $countryDataProvider;
+        $this->tools = $tools;
+        $this->validate = $validate;
+    }
+
+    /**
+     * @param string $isoCode
+     *
+     * @return array
+     *
+     * @throws \PrestaShopException
+     */
+    public function getData($isoCode)
+    {
+        $data = ['has_errors' => true, 'rss' => []];
+        $apiUrl = $this->configuration->get('_PS_API_URL_');
+
+        $blogXMLResponse = $this->circuitBreaker->call($apiUrl . '/rss/blog/blog-' . $isoCode . '.xml');
+
+        if (empty($blogXMLResponse)) {
+            return $data;
+        }
+
+        $rss = @simplexml_load_string($blogXMLResponse);
+        if (!$rss) {
+            return $data;
+        }
+
+        $articles_limit = self::NUM_ARTICLES;
+
+        $shop_default_country_id = (int) $this->configuration->get('PS_COUNTRY_DEFAULT');
+        $shop_default_iso_country = mb_strtoupper($this->countryDataProvider->getIsoCodebyId($shop_default_country_id), 'utf-8');
+        $analytics_params = [
+            'utm_source' => 'back-office',
+            'utm_medium' => 'rss',
+            'utm_campaign' => 'back-office-' . $shop_default_iso_country,
+        ];
+
+        foreach ($rss->channel->item as $item) {
+            if ($articles_limit == 0
+                || !$this->validate->isCleanHtml((string) $item->title)
+                || !$this->validate->isCleanHtml((string) $item->description)
+                || !isset($item->link, $item->title)) {
+                break;
+            }
+            $analytics_params['utm_content'] = 'download';
+            if (in_array($this->contextMode, array(ContextCore::MODE_HOST, ContextCore::MODE_HOST_CONTRIB))) {
+                $analytics_params['utm_content'] = 'cloud';
+            }
+            $article_link = (string) $item->link . '?' . http_build_query($analytics_params);
+
+            $url_query = parse_url($item->link, PHP_URL_QUERY);
+            parse_str($url_query, $link_query_params);
+            if ($link_query_params) {
+                $full_url_params = array_merge($link_query_params, $analytics_params);
+                $base_url = explode('?', (string) $item->link);
+                $base_url = (string) $base_url[0];
+                $article_link = $base_url . '?' . http_build_query($full_url_params);
+            }
```

Comment:
manage the `else` case to integrate `$analytics_params` even if the link has no parameters

Reply:
:+1: